presto-orc/src/main/java/io/prestosql/orc/reader/ColumnReaders.java

```diff
@@ -32,6 +33,9 @@ public static ColumnReader createColumnReader(
             OrcBlockFactory blockFactory)
             throws OrcCorruptionException
     {
+        if (type instanceof TimeType) {
```

Comment:
I realized why this is problematic. In the ORC code, which is generic, we making a huge assumption that the `time` type is written as a `long` with microsecond encoding. We should be able to do this in a safe way by checking the `iceberg.long-type = TIME` attribute from `OrcColumn`. The code would still have this ugly Iceberg specific logic, but it would be safe if we introduce other usages of `TimeType` in the future.

Reply:
I made this change and force-pushed.

Reply:
Verify three things:
* `precision == 6`
* `column.getColumnType() == LONG`
* `"LONG".equals(column.getAttributes("iceberg.long-type"))`